src/clj/rems/api/applications_v2.clj

```diff
@@ -69,32 +80,62 @@
+     (let [raw-apps (reduce all-applications-view (::raw-apps state) events)
+           updated-app-ids (distinct (map :application/id events))
+           ;; TODO: batched injections: only one DB query to fetch all catalogue items etc.
+           ;;       - fetch all items in the background as a batch, use plain maps as injections
+           ;;       - change db/get-license and db/get-user-attributes to fetch all rows if ID is not defined
+           cached-injections (map-vals memoize injections)
+           enriched-apps (->> (select-keys raw-apps updated-app-ids)
+                              (map-vals #(model/enrich-with-injections % cached-injections))
+                              (merge (::enriched-apps state)))]
+       {::raw-apps raw-apps
+        ::enriched-apps enriched-apps
+        ::apps-by-user (group-apps-by-user (vals enriched-apps))}))))
+
+(defn get-all-unrestricted-applications []
+  (-> (refresh-all-applications-cache!)
+      ::enriched-apps
+      (vals)))
 
 (defn get-all-applications [user-id]
-  (let [cache @all-applications-cache
-        events (applications/get-dynamic-application-events-since (:last-processed-event-id cache))
-        applications (reduce all-applications-view (:applications cache) events)
-        ;; TODO: for a shared cache it may be necessary to make assoc-injections idempotent and consider cache invalidation
-        cached-injections (map-vals memoize injections)]
-    (when-let [event-id (:event/id (last events))]
-      (when (compare-and-set! all-applications-cache
-                              cache
-                              {:last-processed-event-id event-id
-                               :applications applications})
-        (log/info "Updated all-applications-cache from" (:last-processed-event-id cache) "to" event-id)))
-    (->> (vals applications)
-         (map #(model/apply-user-permissions % user-id))
-         (remove nil?)
-         (map #(model/enrich-with-injections % cached-injections))
-         (map exclude-unnecessary-keys-from-overview))))
+  (-> (refresh-all-applications-cache!)
+      (get-in [::apps-by-user user-id])
+      (->> (map exclude-unnecessary-keys-from-overview))))
 
 (defn- own-application? [application]
-  (some #{:applicant
-          :member}
-        (:application/roles application)))
+  (some #{:applicant :member} (:application/roles application)))
 
 (defn get-own-applications [user-id]
   (->> (get-all-applications user-id)
        (filter own-application?)))
+
+(defn reload-cache! []
+  (events-cache/empty! all-applications-cache)
+  (refresh-all-applications-cache!))
+
+;; empty the cache occasionally in case some of the injected entities are changed
+(mount/defstate all-applications-cache-reloader
+  :start (doto (ScheduledThreadPoolExecutor. 1)
+           (.scheduleWithFixedDelay reload-cache! 1 1 TimeUnit/HOURS))
+  :stop (doto all-applications-cache-reloader
+          (.shutdown)
```

Comment:
There wasn’t any clear place where to check for interruptions - it would need to be inside the reduced functions. And this is anyways much faster than the pollers, so the 60 second timeout is enough for about 20k applications.